### PR TITLE
String table cleanups

### PR DESCRIPTION
**Pull request by yuwata**
c.f. comments by @poettering https://github.com/systemd/systemd/pull/9685#discussion_r204404704

### CODE REVIEW

**Comment by poettering** [comment]
lgtm!

File: src/resolve/resolved-dns-server.h [hunk 1]
```diff
@@ -13,8 +13,9 @@ typedef enum DnsServerType {
         DNS_SERVER_SYSTEM,
         DNS_SERVER_FALLBACK,
         DNS_SERVER_LINK,
+        _DNS_SERVER_TYPE_MAX,
```
Comment (evverx) [inline src/resolve/resolved-dns-server.h:16]: gcc seems to have started to complain that "enumeration value ‘_DNS_SERVER_TYPE_INVALID’ not handled in switch":
```sh
../src/resolve/resolved-dns-server.c: In function ‘dns_server_unlink’:
../src/resolve/resolved-dns-server.c:147:9: warning: enumeration value ‘_DNS_SERVER_TYPE_MAX’ not handled in switch [-Wswitch]
         switch (s->type) {
         ^~~~~~
../src/resolve/resolved-dns-server.c:147:9: warning: enumeration value ‘_DNS_SERVER_TYPE_INVALID’ not handled in switch [-Wswitch]
```

Comment (yuwata) [reply src/resolve/resolved-dns-server.h:16]: @evverx Sorry. I have no idea why I missed this...
Fixed by #9706.

Comment (keszybz) [reply src/resolve/resolved-dns-server.h:16]: Yeah, a warning in compilation with `-O0` would be pretty annoying → #9707.

**Pull request merged**